pkg/imageMetadata/imageMetadata.go

```diff
@@ -0,0 +1,47 @@
+package imageMetadata
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/rwcarlsen/goexif/exif"
+)
+
+func GetImageCreationDate(imagePath string) (creationDate time.Time, Error error) {
+	file, err := os.Open(imagePath)
+	if err != nil {
+		return time.Time{}, fmt.Errorf("error opening image: %v", err)
+	}
+
+	ExifMetadata, err := exif.Decode(file)
+	if err != nil {
+		return time.Time{}, fmt.Errorf("error decoding exif metadata: %v", err)
+	}
+
+	createdDate, err := ExifMetadata.DateTime()
+	if err != nil {
+		return time.Time{}, fmt.Errorf("error extracting date: %v", err)
+	}
+
+	return createdDate, nil
+}
+
+func GetImageLocation(imagePath string) (latitude float64, longitude float64, Error error) {
```

Comment:
There should be one overarching function which gets all the data I feel like. 

that maybe take the opened image as input. But that is probably premature optimisation. 

There should probably though be a image metadata struct with the information togther